### PR TITLE
draft #2532: refactor and annotate tests in app module

### LINKED ISSUE #2532: Move all tests under app layer to test/
We have local tests under the app module test folder that only run on Robolectric. Now that we have the RunOn annotation, we should prefer keeping all tests under app module's sharedTest folder & instead use the RunOn animation. This issue is tracking:
- Moving all app module tests from the test folder to sharedTest counterparts
- Annotating the tests OR test class with RunOn(ROBOLECTRIC)
- Removing the Gradle/Bazel code around the app module test folder

Update 19 Apr 2025: We should actually isolate to a common test/ folder rather than sharedTest/ and remove sharedTest/ as there's no value to differentiate in the post-Gradle world.

### PR DESCRIPTION
**Pull request by dipu989**
Fixes #2532 (Draft): This PR aims to move the tests from `test` folder to `sharedTest` folder in their respective counterpart's folders.

Checklist for this draft - 
1) Moved all the tests to the intended folders.
2) Annotated the test's class with RunOn(TestPlatform.ROBOELECTRIC)
3) Ran all the tests locally and all of them passed.

Yet to do - 
1) Removing the Gradle/Bazel code around the app module test folder

### CODE REVIEW

File: app/src/sharedTest/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt [hunk 1]
```diff
@@ -75,6 +77,8 @@ import javax.inject.Singleton
   application = AdministratorControlsFragmentTest.TestApplication::class,
   qualifiers = "port-xxhdpi"
 )
+
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt]: Empty line not required

Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt]: Ditto for all

**Review by anandwana001 — CHANGES_REQUESTED** [review]

**Comment by dipu989** [comment]
PTAL if the changes are the intended ones.

File: app/src/sharedTest/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt (hunk 1 above)
Comment (dipu989) [reply app/src/sharedTest/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt]: Sure, will remove them.

**Comment by anandwana001** [comment]
For draft pr, use draft option of GitHub in the right side.
https://github.blog/2019-02-14-introducing-draft-pull-requests/

**Review by anandwana001 — CHANGES_REQUESTED** [review]
LGTM, Thanks @dipu989 

We ca move forward with the last part `Removing the Gradle/Bazel code around the app module test folder`

**Comment by dipu989** [comment]
@anandwana001 Sure, I will update the PR accordingly. Thanks!

**Comment by anandwana001** [comment]
@dipu989 please resolve the conflicts.

**Comment by dipu989** [comment]
@anandwana001 Resolved the conflicts, PTAL!
Roboelectric checks are failing.
Do I need to remove the block of `Unit Test`  [this](https://github.com/oppia/oppia-android/blob/2d8930664b86a98b16ec1d0d98c6f2742ce13baa/app/BUILD.bazel#L715) from Bazel file as they have been moved to `sharedTest` folder now?

**Comment by anandwana001** [comment]
> @anandwana001 Resolved the conflicts, PTAL!
> Roboelectric checks are failing.
> Do I need to remove the block of `Unit Test` [this](https://github.com/oppia/oppia-android/blob/2d8930664b86a98b16ec1d0d98c6f2742ce13baa/app/BUILD.bazel#L715) from Bazel file as they have been moved to `sharedTest` folder now?

I guess, yes. we need to remove this one as nothing will exist there now. Let's commit and see.

**Comment by dipu989** [comment]
@anandwana001 Committed the changes.
I have moved the file `DataBinderMapperImpl.java` from the `test` folder to `sharedTest` folder too in the recent commit.
PTAL!

**Comment by anandwana001** [comment]
@dipu989 looks like we do have checks failing. Could you check on these?

**Comment by anandwana001** [comment]
Closing this PR due to force push, you can make a new one.

**Comment by dipu989** [comment]
@anandwana001 After updating my fork with the latest upstream, I checked out with a new branch and am cherry-picking the commits from this PR's branch. But merge conflicts are coming, which are resolved too in this same PR but with different commits.
So is there any way around so that I can cherry-pick all the commits without fixing merge conflicts again that has already been fixed?

**Comment by anandwana001** [comment]
> which are resolved too in this same PR

But, I think you cannot use the same pr or this branch as we have a force push in this branch.

**Comment by dipu989** [comment]
@anandwana001 Yes, I got this, So I was trying to create a new branch itself and cherry-pick the commis and am facing issues mentioned above. Is there any way I can cherry-pick the commits without fixing the merge commits all over again (as they are already fixed in the subsequent commits)?

**Comment by anandwana001** [comment]
@dipu989  Is it fixed now?

**Comment by dipu989** [comment]
@anandwana001 No, the issue that I stated above still persists.
I might have to fix all the merge conflicts again in every file that I already did before.

**Comment by anandwana001** [comment]
Let me know if you still want to continue on this or you can take other issues to work on.

**Comment by dipu989** [comment]
> 
> 
> Let me know if you still want to continue on this or you can take other issues to work on.

I will try creating a fresh PR will all the changes. If issue still persists, will let you know.

**Pull request closed without merging**